sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/GpuIntervalUtils.scala

```diff
@@ -292,17 +293,15 @@ object GpuIntervalUtils extends Arm {
   // not close firstSignInTable and secondSignInTable here, outer table.close will close them
   private def finalSign(
       firstSignInTable: ColumnVector, secondSignInTable: ColumnVector): ColumnVector = {
-    withResource(Scalar.fromString("-")) { negScalar =>
-      withResource(negScalar.equalTo(firstSignInTable)) { neg1 =>
-        withResource(negScalar.equalTo(secondSignInTable)) { neg2 =>
-          withResource(neg1.bitXor(neg2)) { s =>
-            withResource(Scalar.fromLong(1L)) { one =>
-              withResource(Scalar.fromLong(-1L)) { negOne =>
-                s.ifElse(negOne, one)
-              }
-            }
-          }
-        }
+    val negatives = withResource(Scalar.fromString("-")) { negScalar =>
+      withResource(Seq(firstSignInTable, secondSignInTable).safeMap(negScalar.equalTo)) {
+        case Seq(neg1, neg2) => neg1.bitXor(neg2)
+      }
+    }
+
+    withResource(negatives) { s =>
+      withResource(Scalar.fromLong(1L)) { posOne =>
+        withResource(Scalar.fromLong(-1L))(negOne => s.ifElse(negOne, posOne))
```

Comment:
nit, I'd prefer the use of curly brackets for a `withResource` block: 
```
withResource(x) { negOne =>
  s.ifElse(negOne, posOne)
}
```

Reply:
e.g. treat like `try` or `if`, where scope is something we want to reinforce with indentation.